packages/angular_devkit/build_angular/src/builders/browser-esbuild/index.ts

```diff
@@ -0,0 +1,302 @@
+ * The options are compatible with the Webpack-based builder.
+ * @param options The browser builder options to use when setting up the application build
+ * @param context The Architect builder context object
+ * @returns A promise with the builder result output
+ */
+// eslint-disable-next-line max-lines-per-function
+export async function execute(
+  options: BrowserBuilderOptions,
+  context: BuilderContext,
+): Promise<BuilderOutput> {
+  const startTime = Date.now();
+
+  // Inform user of experimental status of builder and options
+  generateExperimentalWarnings(options, context);
+
+  // Determine project root paths
+  const projectName = context.target?.project;
+  if (!projectName) {
+    context.logger.error(`The 'browser-esbuild' builder requires a target to be specified.`);
+
+    return { success: false };
+  }
+
+  const {
+    workspaceRoot,
+    mainEntryPoint,
+    polyfillsEntryPoint,
+    optimizationOptions,
+    outputPath,
+    sourcemapOptions,
+    tsconfig,
+    assets,
+    outputNames,
+  } = await normalizeOptions(context, projectName, options);
+
+  // Clean output path if enabled
+  if (options.deleteOutputPath) {
+    deleteOutputDir(workspaceRoot, options.outputPath);
+  }
+
+  // Setup bundler entry points
+  const entryPoints: Record<string, string> = {
+    main: mainEntryPoint,
+  };
+  if (polyfillsEntryPoint) {
+    entryPoints['polyfills'] = polyfillsEntryPoint;
+  }
+  // Create reverse lookup used during index HTML generation
+  const entryPointNameLookup = new Map<string, string>();
+  for (const [name, filePath] of Object.entries(entryPoints)) {
+    entryPointNameLookup.set(path.relative(workspaceRoot, filePath), name);
+  }
+
+  // Execute esbuild
+  const result = await bundleCode(
+    workspaceRoot,
+    entryPoints,
+    outputNames,
+    options,
+    optimizationOptions,
+    sourcemapOptions,
+    tsconfig,
+  );
+
+  // Log all warnings and errors generated during bundling
+  if (result.warnings.length) {
+    const warningMessages = await formatMessages(result.warnings, { kind: 'warning', color: true });
+    context.logger.warn(warningMessages.join('\n'));
+  }
+
+  if (result.errors.length) {
+    const errorMessages = await formatMessages(result.errors, { kind: 'error', color: true });
+    context.logger.error(errorMessages.join('\n'));
+  }
+
+  // Return if the bundling failed to generate output files or there are errors
+  if (!result.outputFiles || result.errors.length) {
+    return { success: false };
+  }
+
+  // Structure the bundling output files
+  const initialFiles: FileInfo[] = [];
+  const outputFiles: OutputFile[] = [];
+  for (const outputFile of result.outputFiles) {
+    // Entries in the metafile are relative to the `absWorkingDir` option which is set to the workspaceRoot
+    const relativeFilePath = path.relative(workspaceRoot, outputFile.path);
+    const entryPoint = result.metafile?.outputs[relativeFilePath].entryPoint;
```

Comment:
The TS  option `noUncheckedIndexedAccess` is not used so the type for `outputs[relativeFilePath]` will not contain `undefined` automatically.  It would be safer to add the optional chaining to the end though since technically it might not be present. However, that would be more of an internal error since the outputs names should match the outputFiles.